### PR TITLE
Quality: Remove "reusable block name hint" code

### PR DESCRIPTION
**Pull request by t-hamano**
Related to #52664

## What?

This PR removes the block name hint, and associated code, from the Create Pattern modal.

| Before | After |
|--------|--------|
| before | after | 

## Why?

This hint was added in WordPress 6.3 to let users know that the name changed from "Reusable Block" to "Synced Pattern". As [this comment](https://github.com/WordPress/gutenberg/pull/52664/files#diff-e2733ba5046877c02344a307ea83132978c799a5eb4fd643f612b5daeced3539R14-R16) indicates, this hint was scheduled to be removed in 6.4, so I don't think it's necessary anymore.

## How?

I've removed the relevant components, hooks and CSS. they're private APIs so it should be fine to remove them directly.

## Testing Instructions

- Access `http://localhost:8888/wp-admin/post-new.php?post_type=wp_block`.
- To enable hints, run the following code in your browser console:
  ```
  window.wp.data.dispatch(window.wp.preferences.store).set('core', 'isResuableBlocksrRenameHintVisible', true);
  ```
- No hints will be displayed.

### CODE REVIEW

File: packages/block-editor/README.md [hunk 1]
```diff
@@ -730,10 +730,6 @@ _Returns_
 
 -   `JSX.Element`: A React element.
 
-### ReusableBlocksRenameHint
-
-Undocumented declaration.
-
```
Comment (Mamaduka) [inline packages/block-editor/README.md:736]: Was this a public component?

**Review by Mamaduka — APPROVED** [review]
Makes sense 👍

File: packages/block-editor/README.md (hunk 1 above)
Comment (t-hamano) [reply packages/block-editor/README.md:736]: Oh, I just realized that this component is both private and public at the same time 😅

According to [WPdirectory's results](https://wpdirectory.net/search/01J2RH120VV51K7BTA912YZJD4), there are two plugins other than the Gutenberg plugin that use this component.

Should we keep this component?

Comment (Mamaduka) [reply packages/block-editor/README.md:736]: I wonder if those two plugins are bundling the `block-editor` package. I can't find the `ReusableBlocksRenameHint` instance in their [codebase](https://github.com/search?q=repo%3Amailpoet%2Fmailpoet%20ReusableBlocksRenameHint&type=code).

The same is probably true for the "Kubio AI Page Builder" plugin.

Comment (t-hamano) [reply packages/block-editor/README.md:736]: I downloaded the two plugins. Neither plugin has pre-build code, but looking at the build files, it doesn't look like they use this component.

Therefore, I think it's safe to delete it.

Comment (Mamaduka) [reply packages/block-editor/README.md:736]: Yeah, it should be okay. Let's wait until tomorrow if others have different opinions regarding the public component.

**Comment by Mamaduka** [comment]
Let's merge this.

**Pull request merged**